src/components/VTextField/VTextField.js

```diff
@@ -128,6 +132,23 @@ export default {
   },
 
   methods: {
+    maskedUpdateOnValueChange (val) {
+      const masked = this.maskText(this.unmaskText(val))
+      const value = this.unmaskText(masked)
+
+      this.lazyValue = this.getLazyValue(value)
+      this.$nextTick(() => {
+        this.$refs.input.value = masked
+        this.$emit('input', this.lazyValue)
+      })
+    },
+    updateOnValueChange (val) {
+      this.lazyValue = val
+      this.$emit('input', this.lazyValue)
```

Comment:
This is causing `input` to be emitted twice, as you're also doing it in both `inputValue.set()` and `updateRange()`

Reply:
Well, actually I didn't quite get it when you said:

> Also, this whole function should probably be split by an if statement so none of this casting is happening on non-masked inputs. The same applies for the value watcher too.

I understood it as having if statement to call different functions. Maybe because I didn't understand what you meant about casting issue anywhere in that function. Frankly I also don't like the changes I made and in fact I don't like the new function names I gave. Perhaps you would want to elaborate your suggestion.

Reply:
On multiple emit problem, I realize that the `watch.value` has never been completed for what it's supposed to function as. From the comment, it should handle external change but there is nothing in the function to detect whether the change is external or internal. So, there have always been multiple emits from internally. I'm going to address this in a later PR.

Reply:
I said that because `this.unmaskText(this.maskText(this.unmaskText(val)))` was being run on every change, even on non-masked inputs.

Reply:
> there have always been multiple emits from internally

`<v-text-field :mask="####" v-model.number="num"/>`

Click inside, type `1`, blur. 

On dev:
![image](https://user-images.githubusercontent.com/16421948/31577198-09bc2164-b156-11e7-95f4-467fbb85d9f0.png)

On this branch:
![image](https://user-images.githubusercontent.com/16421948/31577203-27686556-b156-11e7-8215-193bd2467f9b.png)